mythril/laser/plugin/plugins/plugin_annotations.py

```diff
@@ -67,3 +86,30 @@ def __copy__(self):
         result = WSDependencyAnnotation()
         result.annotations_stack = copy(self.annotations_stack)
         return result
+
+    def check_merge_annotation(self, annotation: "WSDependencyAnnotation") -> bool:
+        if len(self.annotations_stack) != len(annotation.annotations_stack):
+            return False
+        for a1, a2 in zip(self.annotations_stack, annotation.annotations_stack):
+            if a1 == a2:
+                continue
+            if type(a1) != type(a2):
+                return False
+            try:
+                if a1.check_merge_annotation(a2) is False:
+                    return False
+            except AttributeError:
+                log.info(
+                    "check_merge_annotation() method doesn't exist "
+                    "for the annotation {}. Aborting merge for the state".format(
+                        type(a1)
+                    )
+                )
+                return False
+        return True
+
+    def merge_annotation(self, annotation: "WSDependencyAnnotation"):
+        for a1, a2 in zip(self.annotations_stack, annotation.annotations_stack):
```

Comment:
What happens if you merge annotations for two states which have seen different amounts of transactions?

I think in this case, their respective annotation stacks will look like:
[a, b]
[c, d, e]

I believe using zip, would ignore `e`

Reply:
nvm, this is captured by the check_merge_annotation 👍